### PR TITLE
 GFX1151 inline asm compilation error

### PR DESCRIPTION
**Pull request by ivandobskygithub**
Encountered the below errors originally running with a local build of MIOpen , running PyTorch wheels to run TTS model training.
Investigation shows similar - https://github.com/ROCm/MIOpen/issues/1528

This is a PR applying similar changes to this PR: https://github.com/ROCm/MIOpen/pull/1531/files to add necessary fixes.

:14:20: error: not a valid operand.
v_add_f32 v3 v3 v3 row_bcast:15 row_mask:0xa
^
:15:23: error: not a valid operand.
v_add_f32 v16 v16 v16 row_bcast:15 row_mask:0xa
^
:17:20: error: not a valid operand.
v_add_f32 v3 v3 v3 row_bcast:31 row_mask:0xc
^
:18:23: error: not a valid operand.
v_add_f32 v16 v16 v16 row_bcast:31 row_mask:0xc
^
MIOpen(HIP): Error [Do] 'amd_comgr_do_action(kind, handle, in.GetHandle(), out.GetHandle())' : ERROR (1)
MIOpen(HIP): Error [BuildOcl] comgr status = ERROR (1)
MIOpen(HIP): Warning [BuildOcl] error: cannot compile inline asm
error: cannot compile inline asm
error: cannot compile inline asm
error: cannot compile inline asm
4 errors generated

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -48,6 +48,7 @@ option( MIOPEN_TEST_GFX900 "Test on Vega10 (gfx900)" OFF )
 option( MIOPEN_TEST_GFX906 "Test on Vega20 (gfx906)" OFF )
 option( MIOPEN_TEST_GFX103X "Test on Navi21/22 (gfx1030/31)" OFF )
 option( MIOPEN_TEST_GFX110X "Test on Navi31/32 (gfx1100/02)" OFF )
+option(MIOPEN_TEST_GFX115X "Test on Navi33 (gfx1150/51)" OFF)
```
Comment (randyspauldingamd) [inline test/CMakeLists.txt]: Formatting

File: src/kernels/MIOpenBatchNormActivBwdPerAct.cl [hunk 2]
```diff
@@ -34,7 +34,7 @@
 #endif
 
 #define MIOPEN_USE_AMDGCN 0
-#if defined(__AMDGCN__) && !(MIO_BN_GFX103X || MIO_BN_GFX110X || MIO_BN_GFX120X)
+#if defined(__AMDGCN__) && !(MIO_BN_GFX103X || MIO_BN_GFX110X || MIO_BN_GFX120X || MIO_BN_GFX115X)
```
Comment (randyspauldingamd) [inline src/kernels/MIOpenBatchNormActivBwdPerAct.cl]: Sort in ascending numeric order. There are many other instances of this, please fix them all.

File: test/CMakeLists.txt
Comment (randyspauldingamd) [inline test/CMakeLists.txt:1]: Could you add comments here: https://github.com/ROCm/MIOpen/blob/ef4a6b4d6610f38d24e34f5d696cc64d3f734850/test/CMakeLists.txt#L558-L560

**Review by randyspauldingamd — CHANGES_REQUESTED** [review]

**Comment by ivandobskygithub** [comment]
Thanks for the feedback, I've incorporated the updates. I've removed the composable kernels changes in this PR for now given the refactoring in the main branch has changed those functions substantially.

File: test/gtest/gtest_common.hpp [hunk 3]
```diff
@@ -138,7 +138,8 @@ enum class Gpu : int
     gfx950  = 1 << 5,
     gfx103X = 1 << 6,
     gfx110X = 1 << 7,
-    gfx120X = 1 << 8,
+    gfx115X = 1 << 8,
+    gfx120X = 1 << 9,
```
Comment (BradPepersAMD) [inline test/gtest/gtest_common.hpp]: It would be more safe to leave the gfx120X value at 1<<8 otherwise I'd have to look at the code more closely to ensure we aren't ever somehow saving or depending on this ordering.

Comment (ivandobskygithub) [reply test/gtest/gtest_common.hpp]: Thanks for the new review, that's a good point. I've pushed an update to set the enum for gfx120 to its previous value, and pointed the gfxlast as gfx115x.

**Comment by BradPepersAMD** [comment]
Running our CI on this includes using clang-format-12 and a few of the changed files are failing this check.  Please ensure these files have been put through clang-format using `clang-format-12 -style=file -i <path-to-source-file>`:

../src/solver/batchnorm/forward_inference.cpp
../src/kernels/batchnorm_functions.h

**Comment by ivandobskygithub** [comment]
> Running our CI on this includes using clang-format-12 and a few of the changed files are failing this check. Please ensure these files have been put through clang-format using `clang-format-12 -style=file -i <path-to-source-file>`:
> 
> ../src/solver/batchnorm/forward_inference.cpp ../src/kernels/batchnorm_functions.h

I've applied formatting with clang-format, but had to use a slightly later version due totaht package not available on my OS. i see some builds still failing around composable kernels.  I had to explicitly disable those when i was building MIOpen.

**Comment by amd-hsivasun** [comment]
Imported to ROCm/rocm-libraries
